dev/io.openliberty.checkpoint_fat/test-applications/jaxrspropagation/src/jaxrspropagation/JaxRsEndpoints.java

```diff
@@ -0,0 +1,396 @@
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.core.Application;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriInfo;
+
+/**
+ * Tests MP Telemetry integration with restfulWS and mpRestClient.
+ * <p>
+ * Whenever a request is made or received by one of these features, MP Telemetry should create a span.
+ * Trace ID information should be sent via the HTTP headers so that the traces from both the client and server can be joined up.
+ * Baggage information should also be sent via the HTTP headers so that contextual information from the first service can be included in trace messages in the second service.
+ * <p>
+ * This class covers:
+ * <p>
+ * Creation of Spans in
+ * <ul>
+ * <li>JAX-RS Client {1}
+ * <li>JAX-RS Server {2}
+ * <li>MP Rest Client {3}
+ * </ul>
+ * <p>
+ * Propagation of Spans from
+ * <ul>
+ * <li>JAX-RS Server to JAX-RS Client {4}
+ * <li>JAX-RS Server to MP Client {5}
+ * <li>JAX-RS Server to JAX-RS Client Async {6}
+ * <li>JAX-RS Server to MP Client Async {7}
+ * </ul>
+ * <p>
+ * Baggage is correctly Propagated:
+ * <ul>
+ * <li>from JAX-RS Client to JAX-RS Server {8}
+ * <li>from MP Client to JAX-RS Server {9}
+ * <li>from JAX-RS Client async to JAX-RS Server {10}
+ * <li>from MP Client async to JAX-RS Server {11}
+ * </ul>
+ * <p>
+ * Correct application of Semantic convention attributes:
+ * <ul>
+ * <li>TCK RestClientSpanTest only checks HTTP_STATUS, HTTP_METHOD, HTTP_SCHEME, HTTP_TARGET, HTTP_URL, see if there are any others which should be present
+ * <li>There are constants defined for each of the attributes in the spec
+ * </ul>
+ * <p>
+ * <strong>NOTE</strong>: Some of the tests in this class could be re-written more simply using {@code FATServlet} and {@code TestSpans}.
+ * However, we've sometimes seen slightly different behaviour for JAX-RS client and MP Rest client depending on whether or not they're called via a JAX-RS resource method.
+ * Therefore we're leaving these tests as they are so that we have some test coverage of calling JAX-RS client via a JAX-RS resource method.
+ */
+@ApplicationPath("")
+@Path("endpoints")
+public class JaxRsEndpoints extends Application {
+
+    private static final Logger LOGGER = Logger.getLogger(JaxRsEndpoints.class.getName());
+
+    public static final String TEST_PASSED = "Test Passed";
+
+    @Inject
+    private InMemorySpanExporter spanExporter;
+
+    @Inject
+    private HttpServletRequest request;
+
+    private Client client;
+
+    @PostConstruct
+    private void openClient() {
+        LOGGER.info("Creating JAX-RS client");
+        client = ClientBuilder.newClient();
+    }
+
+    @PreDestroy
+    private void closeClient() {
+        if (client != null) {
+            client.close();
+            client = null;
+        }
+    }
```

Comment:
Nobody is using this `client` instance.  I think you can delete this.

Reply:
used the client instance in the test.